examples/joke_bot/src/main.rs

```diff
@@ -0,0 +1,268 @@
+use std::{
+    convert::TryInto,
+    error::Error,
+    io,
+    process::exit,
+    time::{Duration, SystemTime},
+};
+
+use client::http_client;
+use ruma::{
+    api::client::r0::{filter::FilterDefinition, message::send_message_event, sync::sync_events},
+    assign, client,
+    events::{
+        room::message::{MessageType, RoomMessageEventContent},
+        AnySyncMessageEvent, AnySyncRoomEvent,
+    },
+    presence::PresenceState,
+    serde::Raw,
+    RoomId, UserId,
+};
+use serde_json::Value as JsonValue;
+use tokio::fs;
+use tokio_stream::StreamExt as _;
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn Error>> {
+    if let Err(e) = run().await {
+        println!("{}", e);
+        exit(1)
+    }
+    Ok(())
+}
+
+type HttpClient = client::http_client::HyperNativeTls;
+type MatrixClient = client::Client<http_client::HyperNativeTls>;
```

Comment:
Should be consistent here and either always use `client::http_client` (and remove the `http_client` import above) or always use `http_client` directly:

```suggestion
type HttpClient = http_client::HyperNativeTls;
type MatrixClient = client::Client<http_client::HyperNativeTls>;
```